### PR TITLE
[FLINK-6884][table] Improve confused exception information.

### PR DESCRIPTION
**Pull request by sunjincheng121**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-6884][table] Improve confused exception information.")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala [hunk 1]
```diff
@@ -718,16 +718,20 @@ abstract class TableEnvironment(val config: TableConfig) {
...
     }
 
     val fieldTypes = schema.physicalFieldTypeInfo
     val fieldNames = schema.physicalFieldNames
 
     // validate requested type
     if (requestedTypeInfo.getArity != fieldTypes.length) {
-      throw new TableException("Arity of result does not match requested type.")
+      throw new TableException(
+        s"Arity[${requestedTypeInfo.getArity}] of result[${requestedTypeInfo}] does not match " +
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala]: Result and requested types need to be switched.

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala [hunk 2]
```diff
@@ -936,32 +936,42 @@ class CodeGenerator(
       resultFieldNames: Seq[String])
     : GeneratedExpression = {
     // initial type check
+    //s"requested: $requestedTypeInfo; Actual: $fType")
     if (returnType.getArity != fieldExprs.length) {
-      throw new CodeGenException("Arity of result type does not match number of expressions.")
+      throw new CodeGenException(
+        s"Arity[${returnType.getArity}] of result type[$returnType}] does not match " +
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala]: `type[$returnType}]` needs a `{` (or `}` should be removed)

**Comment by sunjincheng121** [comment]
HI @fhueske Thanks very much for your reviewing. Update the PR. according your comments.

Best,
SunJincheng

**Comment by wuchong** [comment]
Good job! 
+1 to merge

**Comment by wuchong** [comment]
merging...

**Pull request closed without merging**